cloudify_cli/tests/commands/test_apply.py

```diff
@@ -0,0 +1,282 @@
+    @patch('cloudify_cli.commands.blueprints.upload')
+    def test_upload_blueprint_before_update_with_zip(
+            self,
+            blueprint_upload_mock,
+            *_):
+        self._mock_client_deployment_id(deployment_id=STUB_DEPLOYMENT_ID)
+        apply_command = 'cfy apply --blueprint-path {bl_path} ' \
+                        '--deployment-id {dep_id} -n {bl_file_name} ' \
+                        '-b {bl_id}'.format(
+                            bl_path=SAMPLE_ARCHIVE_PATH,
+                            dep_id=STUB_DEPLOYMENT_ID,
+                            bl_file_name=DEFAULT_BLUEPRINT_FILE_NAME,
+                            bl_id=STUB_BLUEPRINT_ID)
+        self.invoke(apply_command)
+        install_args = blueprint_upload_mock.call_args_list[0][1]
+
+        self.assertEqual(
+            install_args['blueprint_path'].startswith('/tmp'),
+            True
+        )
+        self.assertEqual(
+            install_args['blueprint_id'],
+            STUB_BLUEPRINT_ID
+        )
+
+    @patch('cloudify_cli.commands.blueprints.upload')
+    @patch('cloudify_cli.commands.deployments.add_deployment_labels')
+    @patch('cloudify_cli.commands.deployments.manager_update')
+    def test_apply_call_deployment_update(self,
+                                          deployment_update_mock,
+                                          add_deployment_labels_mock,
+                                          *_):
+        self._mock_client_deployment_id(deployment_id=STUB_DEPLOYMENT_ID)
+        apply_command = 'cfy apply --blueprint-path {bl_path} ' \
+                        '--deployment-id {dep_id} -b {bl_id} ' \
+                        '--inputs={inputs} --deployment-labels {labels}'\
+            .format(bl_path=SAMPLE_BLUEPRINT_PATH,
+                    dep_id=STUB_DEPLOYMENT_ID,
+                    bl_id=STUB_BLUEPRINT_ID,
+                    inputs=SAMPLE_INPUTS_PATH,
+                    labels=TEST_LABELS)
+
+        self.invoke(apply_command)
+
+        deployment_update_mock.assert_called_with(
+            deployment_id=STUB_DEPLOYMENT_ID,
+            blueprint_path=None,
+            inputs={'key1': 'val1', 'key2': 'val2'},
+            reinstall_list=(),
+            skip_install=False,
+            skip_uninstall=False,
+            skip_reinstall=True,
+            ignore_failure=False,
+            install_first=False,
+            preview=False,
+            dont_update_plugins=False,
+            workflow_id=None,
+            force=False,
+            include_logs=True,
+            json_output=False,
+            tenant_name=None,
+            blueprint_id=STUB_BLUEPRINT_ID,
+            visibility='tenant',
+            validate=False,
+            runtime_only_evaluation=False,
+            auto_correct_types=False,
+            reevaluate_active_statuses=False)
+        add_deployment_labels_mock.assert_called_with(
+            labels_list=[{'label_key': 'label_value'}],
```

Comment:
You have `TEST_LABELS` :)

Reply:
`TEST LABELS` is a string, I cant use it here.

Reply:
Right, my bad.